Use the shared Shipyard-based build infrastructure

skitt opened the pull request:
This updates Coastguard to match other Submariner projects:

* copy the common linting, unit testing GHAs
* use the Shipyard base Dapper image
* update the Makefiles
* bump Ginkgo and Gomega to match the rest of Submariner
* revert to bronze1man’s goStrongswanVici, Miguel’s patch was merged
* drop all the hacking/ and scripts/ scripts, they are replaced with
  Shipyard-based features
* update the documentation to match
* fix up the remaining YAML files to pass linting

This disables the end-to-end tests which are totally broken now that
we no longer support KubeFed.

All this is easiest done in a single commit, even though it touches
many different aspects of the project, because of the general
framework assumed by the shared Shipyard base setup.

Signed-off-by: Stephen Kitt <skitt@redhat.com>

mkolesnik commented on .shflags:
```diff
@@ -0,0 +1,1009 @@
+# $Id$
```
This here by accident?

mkolesnik commented on README.md:
```suggestion
make e2e
```

mkolesnik commented on Makefile:
```diff
@@ -1,14 +1,17 @@
-	./.dapper -m bind $@
+.PHONY: $(TARGETS) build images unit validate
 
 e2e: .dapper ./hacking/e2e_subm.sh
```
These targets should be removed

skitt committed:
Use the shared Shipyard-based build infrastructure

This updates Coastguard to match other Submariner projects:

* copy the common linting, unit testing GHAs
* use the Shipyard base Dapper image
* update the Makefiles
* bump Ginkgo and Gomega to match the rest of Submariner
* revert to bronze1man’s goStrongswanVici, Miguel’s patch was merged
* drop all the hacking/ and scripts/ scripts, they are replaced with
  Shipyard-based features
* update the documentation to match
* fix up the remaining YAML files to pass linting

This disables the end-to-end tests which are totally broken now that
we no longer support KubeFed.

All this is easiest done in a single commit, even though it touches
many different aspects of the project, because of the general
framework assumed by the shared Shipyard base setup.

Signed-off-by: Stephen Kitt <skitt@redhat.com>

skitt commented:
@mkolesnik could you update your review?

mkolesnik commented on .github/workflows/unit.yml:
```diff
@@ -0,0 +1,37 @@
+          set -o pipefail
+          make unit 2>&1 | tee artifacts/unittest.log
+
+      - name: Collect artifacts
+        run: |
+          for f in $(find ./pkg -name "*.xml"); do
+            mkdir -p artifacts/$(dirname $f)
+            cp $f -rfp artifacts/$f
+          done
+
+      - name: Upload artifacts
+        uses: actions/upload-artifact@v1
+        with:
+          name: Unit test artifacts
+          path: artifacts
```
Are we even using this?

mkolesnik commented on Dockerfile.dapper:
```diff
@@ -1,51 +1,12 @@
-FROM ubuntu:16.04
-# FROM arm=armhf/ubuntu:16.04
+FROM quay.io/submariner/shipyard-dapper-base:devel
```
```suggestion
ARG BASE_BRANCH
FROM quay.io/submariner/shipyard-dapper-base:${BASE_BRANCH}
```

dfarrell07 commented:
The scope of this PR doesn't seem to be covered by the commit message to me.

skitt commented:
> The scope of this PR doesn't seem to be covered by the commit message to me.

Good point, I’ve updated the commit message (and the PR description).

mangelajo reviewed (approved):
Wow, thanks for the big update!
